.gitlab-ci.yml: split off in more separate jobs

arnout opened the pull request:
The main goal is to minimize the amount of work done in the run-tests job.

arnout committed:
.gitlab-ci.yml: split off tlvf_test in a separate job

This increases parallelism. It also prepares already for running more
unit tests.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
.gitlab-ci.yml: factor out in-prplmesh-builder template

The jobs that run in the prplmesh-builder docker container have some
commonalities, so factor these into a template. More jobs with that
template will be added.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
.gitlab-ci.yml: split off clang-format into a separate job

More parallel jobs gives earlier feedback. Also it gives more
understandable output if it's a separate job (otherwise it's often not
so clear that it's clang-format that is failing).

While we're at it, add the missing git update-index which could cause
the problem to be reported after the build, which is even more
confusing.

Also while we're at it, generate a full patch. This allows someone who
doesn't have clang-format (or a wrong version) to just apply the patch
and retry.

Also while we're at it extends the comments to explain why it's done in
that way.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
framework: tlvf_test: use default logger, to stdout

For unit tests, we're not interested in complex configurable logging.
Therefore, just use the default logger config to standard output.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
.gitlab-ci.yml: remove workaround for capturing tlvf_test log

Since tlvf_test logs to standard output now, we no longer need the
workaround to capture the log file as an artifact.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

tomereli commented on .gitlab-ci.yml:
```diff
@@ -5,15 +5,21 @@
+    # Run clang-format and check there are no modified files by checking git status
+    - ./clang-format.sh
+    - git update-index -q --refresh
+    - git diff-index -p --exit-code HEAD
```
Does this create the patch? If so, what is the patch name? where will it be stored? I think it's worth adding it to the above comment.

tomereli commented on framework/tlvf/test/tlvf_test.cpp:
```diff
@@ -754,7 +754,13 @@ int test_all()
 
 int main(int argc, char *argv[])
 {
-    mapf::Logger::Instance().LoggerInit("tlvf_test");
```
Is it possible to add the below to `LoggerInit()` as an optional argument? It can be useful for other unit tests so it will save us from duplicating this code.

arnout replied:
I instead gave Logger a default config that logs to stdout, in a separate commit.

tomereli reviewed (approved):
LGTM.

arnout committed:
mapf: common: set default logger config to stdout

Some small tools (e.g. unit tests) really don't need the flexible
logging configuration. We just want them to log to standard output.
Having a configuration file for those is unnecessary complexity.

When Logger is instantiated, set default logger config options: enable
all levels, log to standard output, and set a simple format (without
thread ID or component prefix). All these options are overridden by
ToEasyLoggingString(), so if LoggerConfig() gets called later they no
longer apply.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
